blog/content/post/2021-02-21-tigerlake-osf.md

```diff
@@ -0,0 +1,183 @@
+source representatives from Intel like Nataniel DeSimone, Vincent Zimmer, Brian
+Richardson and Isaac Oram. The are often present on various open source
+firmware events on communities, few examples:
+
+* [OSCF2018 Keynote](https://youtu.be/3VVaCOCCiD8)
+* [OSFC2019 Intel Open Platform Enabling Plans](https://www.youtube.com/watch?v=d2aKDVuFaX8)
+* [OSFC2019 Hardening Firmware Components with Host-based Analysis Tools](https://www.youtube.com/watch?v=cd80acwTYLY)
+
+# OSF on Tiger Lake platform
+
+Tiger Lake is the codename of the 11th generation Intel processors. We had the
+pleasure to get the Tiger Lake Reference Validation Platform (RVP) and test the
+available open source firmware options. coreboot development for Tiger Lake
+begun some time ago so that when FSP is released, the build target for Tiger
+Lake RVP should be ready. This however is different from EDK2 MinPlatform. The
+open board implementation is released some time after FSP. But let's start form
+the beginning.
+
+## coreboot
+
+Tiger Lake implementation in coreboot was the first we have tried before going
+with MinPlatform because the latter wasn't available yet. It is quite simple to
+configure the build, since almost all the options are in place when selecting
+the TigerLakeRVP platform. However, one may miss the microcode binary when
+building coreboot. Typically one extract the microcode blob from the original
+BIOS binary shipped with the platform if the microcode is not disclosed or
+publicly available. When extracted simply change the configuration options to
+include microcode external binaries and point to the path with extracted
+microcode, done.
+
+I will omit the build process here since I would like focus more on problems
+and their possible solutions. If you are interested in building coreboot for
+RVP platform please refer to [Booting coreboot on Intel Comet Lake S RVP8 blog post](https://blog.3mdeb.com/2020/2020-08-31-booting-coreboot-on-cometlake/)
+
+The first problem I have encountered is that the platform did not print any
+output on any of the serial consoles, although it is capable of printing it it
+according to the schematics. Thankfully the RVP platforms have 7-segment
+displays for post codes which makes it easy to debug instructions after reset
+vector. It occurred that it stops in the cache as RAM setup. At this point
+there is not much one can do without support or bug history. So I have  sent an
+email to coreboot mailing list and got a reply that older microcode revisions
+had problem with new CAR setup:
+
+https://mail.coreboot.org/hyperkitty/list/coreboot@coreboot.org/thread/7YHWASZX3CQ5U3L7D5CVJCDZSMRNCCXK/
+
+I have followed the advise and turned off the INTEL_CAR_NEM_ENHANCED. I did the
+trick and I could see the serial output on the console. "Now it is a piece of
+cake" I thought... Then I saw the FSP memory init returned an error... Great,
+what now?
+
+Second issue is the memory initialization and configuration. After
```

Comment:
```suggestion
The second challange is the memory initialization and configuration. After
```

Reply:
@miczyg1 maybe each challange should have some fun section title?

Reply:
Applied everywhere to change problem into a challenge. Although I do not see any fun in those challenges and have no idea for any humor here.